test/clojure/nrepl/middleware/print_test.clj

```diff
@@ -231,3 +232,33 @@
                      print/*quota*    8]
              (handle {:value        (range 512)
                       ::print/quota 16}))))))
+
+(deftest print-fn-var-only
+  (testing-print "setting *print-fn* works, but ::print-fn in the message does not"
+    (is (not= (handle {:value (range 10)
+                       ::print/print-fn custom-printer-2})
+              (binding [print/*print-fn* custom-printer-2]
+                (handle {:value (range 10)}))))))
+
+(deftest config-override
+  (testing-print "config file can be used to set default printer"
+    (with-redefs [config/config {::print/print   `custom-printer
+                                 ::print/options {:sub "bar"}}]
+      (is (= (handle {:value          3
+                      ::print/print   `custom-printer
+                      ::print/options {:sub "bar"}})
+             (handle {:value 3})))))
+  (testing-print "config file can be used to set streaming"
+    (with-redefs [config/config {::print/stream? 1
+                                 ::print/buffer-size 8}]
+      (is (= [{:value "(0 1 2 3"}
+              {:value " 4 5 6 7"}
+              {:value " 8 9)"}
+              {}]
+             (handle {:value (range 10)}))))))
+
+(deftest print-utils
```

Comment:
Probably you can submit this test in a separate PR, as it's unrelated to this one.